backend/search-module/services/searchService.ts

```
@@ -1,137 +1,155 @@
-				newStartingPoint = results.docs[results.docs.length - 1]?.data().titleLower;
-			}
-
-			console.log("setting starting point: ", newStartingPoint)
-			return { documents, newStartingPoint };
-		}
-	};
+  private static algoliaClient: ReturnType<typeof algoliasearch> | null = null;
+  private static readonly ALGOLIA_INDEX_NAME = "contents";
+
+  private static getAlgoliaClient() {
+    if (!this.algoliaClient) {
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)